testing/btest/coverage/record-fields.test

```diff
@@ -0,0 +1,55 @@
+# @TEST-DOC: Output interesting record types in bare and default mode recursively. Currently just the connection record type.
+#
+# @TEST-REQUIRES: ${SCRIPTS}/have-spicy
+# @TEST-EXEC: zeek -b %INPUT >out.bare
+# @TEST-EXEC: TEST_DIFF_CANONIFIER=$SCRIPTS/diff-remove-abspath btest-diff out.bare
+# @TEST-EXEC: zeek %INPUT >out.default
+# @TEST-EXEC: TEST_DIFF_CANONIFIER=$SCRIPTS/diff-remove-abspath btest-diff out.default
+#
+
+global record_types_seen: set[string];
+
+function render_record_type(name: string, rt: any, indent: string)
+	{
+	print fmt("%s%s {", indent, name);
+	local field_names: vector of string;
+	local fields = record_fields(rt);
+
+	for ( fn, _ in fields )
+		field_names += fn;
+	sort(field_names, strcmp);
+
+	for ( _, fn in field_names )
```

Comment:
TBH, I find this implementation which prints while recursively iterating pretty opaque and read-only. Moving that recursive part to a `record_type_to_string(rt: any): string` and collecting that from here might be more maintainable, e.g.,

```zeek
local fields_: table[string] of string;
for ( [fn], fr in record_fields(rt) )
	{ # ...
```

Not a blocker since this seems to do its job, but IMO still nice to have.

Reply:
I've extracted out one helper to determine the nested record type names, but given all the indenting the direct prints were easier.